src/api/slice.cc

```diff
@@ -40,6 +40,12 @@ void slice(const FunctionCallbackInfo<Value>& args) {
     auto vSubject = args[2];
     int sliceStart = args[3]->IntegerValue(context).FromJust();
 
+    int len =  v8::Local<v8::String>::Cast(vResult)->Length();
+    if (len == 0) {
+        args.GetReturnValue().Set(vResult);
+        return;
+    }
```

Comment:
If we change the order, we are not going to cast sliceStart when it is not necessary.
```suggestion

    int len =  v8::Local<v8::String>::Cast(vResult)->Length();
    if (len == 0) {
        args.GetReturnValue().Set(vResult);
        return;
    }

    int sliceStart = args[3]->IntegerValue(context).FromJust();
```